[krunkit, Apple M4 Max]`llama.cpp` cannot be built with an example script on the website: `cc1: sorry, unimplemented: no support for ‘sme’ without ‘sve2’`

`llama.cpp` cannot be built with the following script, perhaps because of the CPU (M4 Max)

https://github.com/lima-vm/lima/blob/7f46ac6a3e257d54857049038a80b0fd40dbf648/website/content/en/docs/config/vmtype/krunkit.md?plain=1#L108-L160

Host:
```console
$ limactl start --vm-type krunkit --cpus 32 --memory 32 template:fedora
```

Guest:
```console
$ sudo ./a.sh
[...]
HEAD is now at 97340b4c9 Vulkan: Don't default to CPU device (like llvmpipe), even if no other device is available, to allow fallback to CPU backend (#14099)
-- The C compiler identification is GNU 15.2.1
-- The CXX compiler identification is GNU 15.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/sbin/git (found version "2.51.1")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- CMAKE_SYSTEM_PROCESSOR: aarch64
-- GGML_SYSTEM_ARCH: ARM
-- Including CPU backend
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- ARM detected
-- Performing Test GGML_COMPILER_SUPPORTS_FP16_FORMAT_I3E
-- Performing Test GGML_COMPILER_SUPPORTS_FP16_FORMAT_I3E - Failed
-- ARM -mcpu not found, -mcpu=native will be used
-- Performing Test GGML_MACHINE_SUPPORTS_dotprod
-- Performing Test GGML_MACHINE_SUPPORTS_dotprod - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_nodotprod
-- Performing Test GGML_MACHINE_SUPPORTS_nodotprod - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_i8mm
-- Performing Test GGML_MACHINE_SUPPORTS_i8mm - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_noi8mm
-- Performing Test GGML_MACHINE_SUPPORTS_noi8mm - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_sve
-- Performing Test GGML_MACHINE_SUPPORTS_sve - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_nosve
-- Performing Test GGML_MACHINE_SUPPORTS_nosve - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_sme
-- Performing Test GGML_MACHINE_SUPPORTS_sme - Failed
-- Performing Test GGML_MACHINE_SUPPORTS_nosme
-- Performing Test GGML_MACHINE_SUPPORTS_nosme - Failed
cc1: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
CMake Warning at ggml/src/ggml-cpu/CMakeLists.txt:163 (message):
  Failed to get ARM features
Call Stack (most recent call first):
  ggml/src/CMakeLists.txt:310 (ggml_add_cpu_backend_variant_impl)


-- Adding CPU backend variant ggml-cpu: -mcpu=native 
-- Found Vulkan: /lib64/libvulkan.so (found version "1.4.321") found components: glslc missing components: glslangValidator
-- Vulkan found
-- GL_KHR_cooperative_matrix supported by glslc
-- GL_NV_cooperative_matrix2 supported by glslc
-- GL_EXT_integer_dot_product supported by glslc
-- GL_EXT_bfloat16 supported by glslc
-- Including Vulkan backend
-- Found CURL: /usr/lib64/libcurl.so (found version "8.15.0")
-- Configuring done (1.0s)
-- Generating done (0.1s)
-- Build files have been written to: /root/llama.cpp/build
[  0%] Generating build details from Git
[  1%] Building C object examples/gguf-hash/CMakeFiles/xxhash.dir/deps/xxhash/xxhash.c.o
[  2%] Creating directories for 'vulkan-shaders-gen'
[  2%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml.c.o
[  3%] Building C object examples/gguf-hash/CMakeFiles/sha256.dir/deps/sha256/sha256.c.o
[  3%] Building C object examples/gguf-hash/CMakeFiles/sha1.dir/deps/sha1/sha1.c.o
[  4%] Building CXX object tools/mtmd/CMakeFiles/llama-llava-cli.dir/deprecation-warning.cpp.o
[  4%] Building CXX object tools/mtmd/CMakeFiles/llama-gemma3-cli.dir/deprecation-warning.cpp.o
-- Found Git: /usr/sbin/git (found version "2.51.1")
[  4%] No download step for 'vulkan-shaders-gen'
[  4%] No update step for 'vulkan-shaders-gen'
[  5%] No patch step for 'vulkan-shaders-gen'
[  5%] Building CXX object common/CMakeFiles/build_info.dir/build-info.cpp.o
[  5%] Performing configure step for 'vulkan-shaders-gen'
[  5%] Built target build_info
[  5%] Building CXX object tools/mtmd/CMakeFiles/llama-minicpmv-cli.dir/deprecation-warning.cpp.o
-- The C compiler identification is GNU 15.2.1
[  5%] Built target sha1
[  5%] Linking CXX executable ../../bin/llama-llava-cli
[  5%] Linking CXX executable ../../bin/llama-gemma3-cli
[  6%] Building CXX object tools/mtmd/CMakeFiles/llama-qwen2vl-cli.dir/deprecation-warning.cpp.o
[  6%] Built target sha256
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml.cpp.o
-- The CXX compiler identification is GNU 15.2.1
-- Detecting C compiler ABI info
[  6%] Built target llama-llava-cli
[  6%] Built target llama-gemma3-cli
[  7%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-alloc.c.o
[  7%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-backend.cpp.o
-- Detecting C compiler ABI info - done
[  8%] Linking CXX executable ../../bin/llama-minicpmv-cli
-- Check for working C compiler: /usr/sbin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
[  8%] Linking CXX executable ../../bin/llama-qwen2vl-cli
[  8%] Built target llama-minicpmv-cli
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-opt.cpp.o
[  9%] Built target llama-qwen2vl-cli
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-threading.cpp.o
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[  9%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-quants.c.o
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Enabling coopmat glslc support
-- Enabling coopmat2 glslc support
-- Enabling dot glslc support
-- Enabling bfloat16 glslc support
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /root/llama.cpp/build/ggml/src/ggml-vulkan/vulkan-shaders-gen-prefix/src/vulkan-shaders-gen-build
[ 10%] Performing build step for 'vulkan-shaders-gen'
[ 50%] Building CXX object CMakeFiles/vulkan-shaders-gen.dir/vulkan-shaders-gen.cpp.o
[ 11%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/gguf.cpp.o
[100%] Linking CXX executable /root/llama.cpp/build/bin/vulkan-shaders-gen
[ 11%] Built target xxhash
[100%] Built target vulkan-shaders-gen
[ 11%] Built target vulkan-shaders-gen-build
[ 11%] Performing install step for 'vulkan-shaders-gen'
-- Install configuration: ""
[ 11%] Built target vulkan-shaders-gen-install
[ 11%] Completed 'vulkan-shaders-gen'
[ 11%] Built target vulkan-shaders-gen
[ 11%] Linking CXX shared library ../../bin/libggml-base.so
[ 11%] Built target ggml-base
[ 11%] Generate vulkan shaders
ggml_vulkan: Generating and compiling shaders to SPIR-V
[ 12%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.cpp.o
[ 12%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.c.o
[ 13%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/hbm.cpp.o
[ 13%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/quants.c.o
[ 13%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/repack.cpp.o
[ 14%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/traits.cpp.o
[ 14%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/amx.cpp.o
cc1: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
cc1plus: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
cc1plus: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
cc1plus: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
cc1: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:79: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:93: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.cpp.o] Error 1
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:107: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/repack.cpp.o] Error 1
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:121: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/hbm.cpp.o] Error 1
cc1plus: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:135: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/quants.c.o] Error 1
cc1plus: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:149: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/traits.cpp.o] Error 1
gmake[2]: *** [ggml/src/CMakeFiles/ggml-cpu.dir/build.make:163: ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/amx.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2196: ggml/src/CMakeFiles/ggml-cpu.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 14%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o
[ 15%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan-shaders.cpp.o
[ 15%] Linking CXX shared library ../../../bin/libggml-vulkan.so
[ 15%] Built target ggml-vulkan
gmake: *** [Makefile:146: all] Error 2
```


Version:
- Lima v2.0.0-rc.0
- krunkit v1.1.1 (libkrun-efi v1.16.0)
- macOS 26.0.1
- MacBook Pro 2024 (Apple M4 Max, 128 GiB)

--- a/website/content/en/docs/config/vmtype/krunkit.md
+++ b/website/content/en/docs/config/vmtype/krunkit.md
@@ -145,11 +145,7 @@ dnf install -y git cmake clang curl-devel glslc vulkan-devel virglrenderer
   git clone https://github.com/ggml-org/llama.cpp
   (
     cd llama.cpp
-    git reset --hard 97340b4c9924be86704dbf155e97c8319849ee19
-    cmake -B build -DGGML_VULKAN=ON -DGGML_CCACHE=OFF -DCMAKE_INSTALL_PREFIX=/usr
-    # FIXME: the build seems to fail on Apple M4 Max (and probably on other processors too).
-    # Error:
-    #   cc1: sorry, unimplemented: no support for ‘sme’ without ‘sve2’
+    cmake -B build -DGGML_VULKAN=ON -DGGML_CCACHE=OFF -DGGML_NATIVE=OFF -DCMAKE_INSTALL_PREFIX=/usr
     cmake --build build --config Release -j8
     cmake --install build
   )

